TA-puppet-alert-actions/bin/inputs/orchestrator_actions.py

```diff
@@ -0,0 +1,135 @@
+#
+# 
+# Build a dictionary of the custom inputs utilized to retrieve the available actions.
+inputs = {}
+inputs['hec_token'] = helper.get_customized_setting("splunk_hec_token")
+inputs['hec_url'] = helper.get_customized_setting("splunk_hec_url")
+inputs['pe_console'] = helper.get_customized_setting("puppet_enterprise_console")
+inputs['timeout'] = helper.get_customized_setting("timeout")
+#
+#
+# Check if there is a configured timeout.
+try:
+  if inputs['timeout'] is not None and inputs['timeout'] is not '':
+    timeout = inputs['timeout']
+  else:
+    timeout = 360
+  #
+  # Retrieve any user accounts configured within the add-on.
+  config_file = os.path.join(SPLUNK_APPS, APP_NAME, 'local/ta_puppet_alert_actions_account.conf')
+  account_list = pie.util.list_accounts(config_file)
+  for user in account_list:
+    inputs['account'] = helper.get_credential_by_username(user)
+    #
+    # Check if the user is configured with an RBAC token or Password:
+    _token = bool(int(inputs['account']['pe_token']))
+    if _token:
+      token = inputs['account']['password']
+    else:
+      user = inputs['account']['username']
+      upass = inputs['account']['password']
+      endpoints = pie.util.getendpoints(inputs['pe_console'])
+      rbac_url = endpoints['rbac']
+      token = pie.rbac.genauthtoken(user,upass,'TA-puppet-alert-actions',rbac_url,timeout)
+    try:
+      #
+      # Build event message for available Tasks by user.
+      tasklist = pie.bolt.get_tasklist(inputs['pe_console'],token)
+      for task in tasklist:
+        t_id = task['id']
+        task_parameter = pie.bolt.get_actionparams(t_id,token)
+        _tmessage = {
+          'permitted': task['permitted'],
+          'task_meta': None,
+          'task_name': task['name'],
+          'task_params': [],
+          'user': user,
+        }
+        tmessage = _tmessage
+        #
+        # Add any task parameters to the event message.
+        try:
+          tmessage['task_meta'] = task_parameter['metadata']['parameters']
+          for param in tmessage['task_meta']:
+            tmessage['task_params'].append(param)
+          #
+          # Post Tasks via HEC endpoint
+          try:
+            host = inputs['pe_console'].replace("https://", "")
+            hec_url = inputs['hec_url']
+            hec_token = inputs['hec_token']
+            post_tasks = pie.hec.post_action(tmessage,host,hec_url,hec_token)
+          except Exception as e:
+            sys.stderr.write("TA-puppet-alert-actions: Failed to post task data to Splunk - {}".format(e))
+        except Exception as e:
+          sys.stderr.write("TA-puppet-alert-actions: Unable to determine task parameters - {}".format(e))
+      #
+      # Build event message for available Plans by user.
+      planlist = pie.bolt.get_planlist(inputs['pe_console'],token)
+      for plan in planlist:
+        p_id = plan['id']
+        plan_parameter = pie.bolt.get_actionparams(p_id,token)
+        _pmessage = {
```

Comment:
are we building a dictionary `_pmessage` and then immediately assigning it to a new variable, `pmessage`, leaving this one unused. seems like an unnecessary extra step?

Reply:
Confirmed this still works without the redundant `_message`.